stix_shifter_modules/darktrace/stix_transmission/connector.py

```diff
@@ -0,0 +1,197 @@
+        except InvalidAuthenticationException:
+            response_dict['code'] = 1001
+            response_dict['message'] = "Invalid Authentication"
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+        except ConnectionError:
+            response_dict['code'] = 1003
+            response_dict['message'] = "Invalid Host/Port"
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+        except Exception as ex:
+            response_dict['type'] = ex.__class__.__name__
+            response_dict['message'] = ex
+            self.logger.error('error when getting search results: %s', ex)
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+        return return_obj
+
+    def get_results_data(self, response_dict):
+
+        proto_field = {"dhcp": "udp", "x509": "ssl", "files_identified": "tcp", "software": "tcp",
+                       "device_details": "udp"}
+
+        results = []
+        for log in response_dict['hits']['hits']:
+            if log['_source']['@type'] in self.config_map['DT_Protocols']:
+                element_dict = {}
+
+                if log['_source']['@type'] != 'conn':
+                    data = {}
+                    for key, value in log['_source']['@fields'].items():
+                        if key in self.config_map['DT_NetworkTraffic']:
+                            data[key] = value
+
+                    if 'proto' not in data.keys():
+                        data['proto'] = proto_field.get(log['_source']['@type'], log['_source']['@type'])
+                    element_dict['conn'] = data
+                element_dict[log['_source']['@type']] = log['_source']['@fields']
+                results.append(element_dict)
+
+        return results
+
+    @staticmethod
+    def check_empty_data(response):
+        """
+        Adding protocol, file name value and remove empty data
+        params: res_dict(dict) response dictionary
+        return: res_dict(dict)
+        """
+
+        file_fields = ["filename", "sha1_file_hash", "sha1", "md5_file_hash", "md5", "sha256_file_hash", "sha256",
+                       "mime", "file_mime_type", "mime_type", "dcc_mime_type", "total_bytes", "file_msg", "read_size",
+                       "write_size", "dcc_file_size"]
+
+        for index in range(len(response['data'])):
+            for key in response['data'][index]:
+                if response['data'][index][key].get("filename") is None:
+                    if response['data'][index][key].keys() & file_fields:
+                        response['data'][index][key]["filename"] = "null"
+
+                empty_keys = [k for k, v in response['data'][index][key].items() if not v]
+                for empty_key in empty_keys:
+                    del response['data'][index][key][empty_key]
```

Comment:
you don't need to remove empty fields. its already handled by json to stix translator

Reply:
Removed code as suggested.